### PR TITLE
build(babel): Upgrade to v7

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Supersedes https://github.com/getsentry/sentry/pull/9666

Pretty straightforward actually

 - We bump jest / babel-jest to support babel.config.js
 - We bump eslint to support babel-eslint
 - We bump node to 8.10.0 to support eslint@5

### CODE REVIEW

File: src/sentry/static/sentry/app/api.jsx [hunk 1]
```diff
@@ -59,13 +58,13 @@ export class Client {
    * If so, redirect user to new project slug
    */
   hasProjectBeenRenamed(response) {
-    let code = response && idx(response, _ => _.responseJSON.detail.code);
+    let code = response?.responseJSON?.detail?.code;
```
Comment (markstory) [inline src/sentry/static/sentry/app/api.jsx]: 😍 Do we start using es features when they hit stage 2? [optional chaining operators](https://github.com/TC39/proposal-optional-chaining) hasn't gone past stage 2 yet.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/api.jsx]: Mostly following @billyvg's lead here https://github.com/getsentry/sentry/pull/9666

File: .storybook/babel.config.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/*eslint-env node*/
+module.exports = {
```
Comment (billyvg) [inline .storybook/babel.config.js]: should we import and use our main babel config?

File: .nvmrc [hunk 3]
```diff
@@ -1 +1 @@
-8.9.1
+8.10.0
```
Comment (billyvg) [inline .nvmrc]: Should we use 8.15.0?

**Review by billyvg — COMMENTED** [review]
Let's make sure this deploys to staging like what you did w/ webpack.

File: .nvmrc (hunk 3 above)
Comment (EvanPurkhiser) [reply .nvmrc]: This was the smallest bump. We could probably go higher if we wanted, but 🤷‍♂️

File: .storybook/babel.config.js (hunk 2 above)
Comment (EvanPurkhiser) [reply .storybook/babel.config.js]: yes, good call

**Review by billyvg — APPROVED** [review]

**Pull request merged**